### ISSUE TITLE
Fix recurring task mask

### ISSUE THREAD

**Issue by Fandekasp** [GothenburgBitFactory/taskwarrior#2157]
I have issues with my recurring tasks, which don't get recreated consistently.
I learnt about [How recurrence works](https://taskwarrior.org/docs/recurrence.html#recurrence), and I'm still a bit unclear about masks, but it looks like the mask was completed (+++), resulting in no new task being created?
This is confusing, since I didn't set an `until:` date (or I maybe added one first, then removed it after understanding that it was unwanted).

I have 2 recurring tasks: one to review my Anki cards, the other one to write a journal entry.

Following #701:

```console
$ task diag

task 2.6.0
   Platform: Linux

Compiler
    Version: 8.3.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      Built: Apr 29 2019 02:17:09
      CMake: 3.14.3
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.6.7
 Build type: 

Configuration
       File: /home/dori/.taskrc (found), 2636 bytes, mode 100644
       Data: /home/dori/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
    $EDITOR: vim
     Server: 
      Trust: strict
Certificate: , not readable, 0 bytes
        Key: , not readable, 0 bytes
    Ciphers: NORMAL
      Creds: 

Hooks
     System: Enabled
   Location: /home/dori/.task/hooks
             (-none-)

Tests
      $TERM: screen-256color (136x64)
       Dups: Scanned 64 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 64 tasks for broken references:
             No broken references found

$ grep 'recur:"' .task/pending.data        
[description:"Complete Anki reviews" due:"1556636399" entry:"1556352551" mask:"+++" modified:"1556585225" project:"BAAS" recur:"daily" scheduled:"1556560800" status:"recurring" uuid:"dbc6c685-e267-4856-a221-74391e197dfd"]
[description:"write jrnl entry" due:"1556636399" entry:"1556521697" imask:"4" modified:"1556521884" parent:"98ae621d-21ca-4663-a942-a2d05180bee6" project:"BAAS" recur:"daily" scheduled:"1556560800" status:"pending" uuid:"1a2f2af1-8a3a-4e12-ae4d-2a03db624b71"]

$ task show recurrence.limit
                       
Config Variable  Value                         
recurrence.limit 2    
  Default value  1    

$ task show report.recurring.filter        
Config Variable         Value
report.recurring.filter (status:pending or status:waiting) and (+PARENT or +CHILD)
          
$ task status:recurring all

ID St UUID     Age Project R Due        Description          
 6 R  dbc6c685 2d  BAAS    R 2019-04-30 Complete Anki reviews

1 task

$ task ready 

ID Age Project  R S   Due Description                               Urg 
 7 17h BAAS     R  7h     write jrnl entry                          14.5

$ task 6 info

Name          Value                                                                 
ID            6
Description   Complete Anki reviews                                                 
Status        Recurring
Project       BAAS                                                                  
Recurrence    daily
Mask          +++                                                                   
Entered       2019-04-27 17:09:11 (2d)
Scheduled     2019-04-30 03:00:00                                                   
Due           2019-04-30 23:59:59
Last modified 2019-04-30 09:47:05 (16min)                                           
Virtual tags  DUE DUETODAY MONTH PARENT SCHEDULED TODAY UNBLOCKED WEEK YEAR PROJECT
UUID          dbc6c685-e267-4856-a221-74391e197dfd                                  
Urgency       14.55

    project        1 *    1 =      1
    scheduled      1 *    5 =      5
    due        0.711 *   12 =   8.53
    age        0.005 *    2 =  0.011
                              ------
                               14.55

Date                Modification                                                          
2019-04-27 17:09:11 Mask set to 'WW'.
2019-04-27 22:33:17 Mask changed from 'WW' to '+W'.                                       
2019-04-28 21:05:43 Mask changed from '+W' to '+WW'.
2019-04-28 21:09:08 Mask changed from '+WW' to '+W-'.                                     
2019-04-28 21:09:10 Mask changed from '+W-' to '+--'.
2019-04-28 21:09:10 Scheduled set to '2019-04-29 00:00:00'.                               
                    Due changed from '2019-04-27 00:00:00' to '2019-04-29 00:00:00'.      
2019-04-28 21:28:57 Mask changed from '+--' to '++-'.
2019-04-28 21:29:02 Mask changed from '++-' to '+++'.                                     
2019-04-30 09:25:07 Wait deleted.
                    Due changed from '2019-04-29 00:00:00' to '2019-04-30 23:59:59'.
                    Scheduled changed from '2019-04-29 00:00:00' to '2019-04-30 03:00:00'.
2019-04-30 09:46:32 Description changed from 'Complete Anki reviews' to 'iMask:0'.        
2019-04-30 09:47:05 Description changed from 'iMask:0' to 'Complete Anki reviews'.

$ task 7 info

Name          Value                                                                           
ID            7
Description   write jrnl entry                                                                
Status        Pending
Project       BAAS                                                                            
Recurrence    daily
Parent task   98ae621d-21ca-4663-a942-a2d05180bee6                                            
Mask Index    4
Entered       2019-04-29 16:08:17 (17h)                                                       
Scheduled     2019-04-30 03:00:00
Due           2019-04-30 23:59:59                                                             
Last modified 2019-04-29 16:11:24 (17h)
Virtual tags  CHILD DUE DUETODAY MONTH PENDING READY SCHEDULED TODAY UNBLOCKED WEEK YEAR      
              PROJECT                                                                         
UUID          1a2f2af1-8a3a-4e12-ae4d-2a03db624b71
Urgency       14.53                                                                           

    project        1 *    1 =      1
    scheduled      1 *    5 =      5
    due        0.711 *   12 =   8.53
                              ------
                               14.53

Date                Modification                                                    
2019-04-29 16:10:43 Scheduled set to '2019-04-30 03:00:00'.
                    Due changed from '2019-04-30 00:00:00' to '2019-04-30 23:59:59'.
2019-04-29 16:11:24 Wait set to '2019-04-30 03:00:00'.                              
                    Status changed from 'pending' to 'waiting'.    

$ task 98ae621d-21ca-4663-a942-a2d05180bee6  info

Name          Value                                      
ID            -
Description   write jrnl entry                           
Status        Deleted
Project       BAAS                                       
Recurrence    daily
Mask          --++W                                      
Entered       2019-04-26 23:05:10 (3d)
Scheduled     2019-04-26 03:00:00                        
Due           2019-04-26 23:59:59
End           2019-04-29 16:12:07                        
Last modified 2019-04-29 22:36:40 (11h)
Virtual tags  DELETED PARENT SCHEDULED UNBLOCKED PROJECT 
UUID          98ae621d-21ca-4663-a942-a2d05180bee6
Urgency       16.38                                      

    project        1 *    1 =      1
    scheduled      1 *    5 =      5
    due        0.864 *   12 =   10.4
    age        0.008 *    2 =  0.016
                              ------
                               16.38

Date                Modification                                                    
2019-04-26 23:05:10 Mask set to '--'.
2019-04-26 23:11:30 Wait set to '2019-04-25 04:00:00'.                              
2019-04-26 23:12:46 Until set to '2019-04-25 04:00:00'.
2019-04-29 16:08:17 Until deleted.                                                  
2019-04-29 16:08:17 Mask changed from '--' to '--WWW'.
2019-04-29 16:09:00 Mask changed from '--WWW' to '--+WW'.                           
2019-04-29 16:09:55 Wait deleted.
                    Scheduled set to '2019-04-26 03:00:00'.
                    Due changed from '2019-04-26 00:00:00' to '2019-04-26 23:59:59'.
2019-04-29 16:11:47 Wait set to '2019-04-26 03:00:00'.                              
                    Status changed from 'recurring' to 'waiting'.                   
2019-04-29 16:12:07 End set to '2019-04-29 16:12:07'.
                    Status changed from 'pending' to 'completed'.
2019-04-29 20:38:54 Status changed from 'completed' to 'deleted'.                   
2019-04-29 22:36:40 Mask changed from '--+WW' to '--++W'.

```

Just completed the Anki reviews, and not seeing any new task scheduled in the future (with `recurrence.limit 2`,  I'd expect to see at least tomorrow's task).

Is there a way to rewrite the template task' mask (if that is the solution)? Can someone help me fix this recurring mess? :)

**Comment by Fandekasp** [comment]
@ddeimeke @jcrben @pbeckingham do you need other information to help me?

Again yesterday, I tried recreating the recurrent task, and again today, it doesn't appear in my task list:

```console
$ task add "Write Journal entry" due:2019-05-06T21:30 rc.dateformat:Y-M-DTH:N recur:daily project:BAAS scheduled:due-20h
```

**Comment by pbeckingham** [comment]
Working on it.

**Comment by pbeckingham** [comment]
Observations (no solution/explanation yet), and questions:

- Don't use 2.6.0, it's not only unreleased, but has lots of test failures in addition to a half-completed rewrite of recurrence.  Yes, recurrence.  Stick to the stable 2.5.1, or the stable-ish almost released 2.5.2.
- This format `Y-M-DTH:N` is built-in, and always recognized, so doesn't need to be specified for input.
- Your `id` 4 task has a mask of `+X`. You created the task on 2019-05-06, and the first child task (`+`) has been completed (marked done), and the second (`X`) deleted.  Did you complete one and delete the second?  Or did that happen without your intervention?
- I expect your task `id` 4 to spawn another child task due on 2019-05-08. Could you confirm/deny that, in the form of another `task 33b6121b  info` tomorrow.

**Comment by Fandekasp** [comment]
* I installed 2.6 for the weekly/daily reports, but ended up adding my custom report, so yes, I'll go back to 2.5.1, thanks!
* format: Good to know, thanks!
* deleted task: I had set `recurrence.limit=2` in my taskrc, and both tasks got created with a due date for yesterday. I was hoping one to be due yesterday and the next one due today (and scheduled early in the morning. I understand better now.
* totally makes sense :+1:
